src/ios/ShareExtension/ShareViewController.m

```diff
@@ -57,231 +57,243 @@ @implementation ShareViewController
-            void (^completion)(BOOL success) = ^void(BOOL success) {
-                NSLog(@"Completions block: %i", success);
-            };
-            if (@available(iOS 13.0, *)) {
-                UISceneOpenExternalURLOptions * options = [[UISceneOpenExternalURLOptions alloc] init];
-                options.universalLinksOnly = false;
-                
-                [invocation setTarget: responder];
-                [invocation setSelector: selector];
-                [invocation setArgument: &url atIndex: 2];
-                [invocation setArgument: &options atIndex:3];
-                [invocation setArgument: &completion atIndex: 4];
-                [invocation invoke];
-                break;
-            } else {
-                NSDictionary<NSString *, id> *options = [NSDictionary dictionary];
-                
-                [invocation setTarget: responder];
-                [invocation setSelector: selector];
-                [invocation setArgument: &url atIndex: 2];
-                [invocation setArgument: &options atIndex:3];
-                [invocation setArgument: &completion atIndex: 4];
-                [invocation invoke];
-                break;
-            }
-        }
-    }
+	SEL selector = NSSelectorFromString(@"openURL:options:completionHandler:");
+
+	UIResponder* responder = self;
+	while ((responder = [responder nextResponder]) != nil) {
+		NSLog(@"responder = %@", responder);
+		if([responder respondsToSelector:selector] == true) {
+			NSMethodSignature *methodSignature = [responder methodSignatureForSelector:selector];
+			NSInvocation *invocation = [NSInvocation invocationWithMethodSignature:methodSignature];
+
+			// Arguments
+			void (^completion)(BOOL success) = ^void(BOOL success) {
+				NSLog(@"Completions block: %i", success);
+			};
+			if (@available(iOS 13.0, *)) {
+				UISceneOpenExternalURLOptions * options = [[UISceneOpenExternalURLOptions alloc] init];
+				options.universalLinksOnly = false;
+
+				[invocation setTarget: responder];
+				[invocation setSelector: selector];
+				[invocation setArgument: &url atIndex: 2];
+				[invocation setArgument: &options atIndex:3];
+				[invocation setArgument: &completion atIndex: 4];
+				[invocation invoke];
+				break;
+			} else {
+				NSDictionary<NSString *, id> *options = [NSDictionary dictionary];
+
+				[invocation setTarget: responder];
+				[invocation setSelector: selector];
+				[invocation setArgument: &url atIndex: 2];
+				[invocation setArgument: &options atIndex:3];
+				[invocation setArgument: &completion atIndex: 4];
+				[invocation invoke];
+				break;
+			}
+		}
+	}
 }
 
-- (void)viewWillAppear:(BOOL)animated {
-    if( [(NSString*)self.contentText length] == 0 ) {
-        [super viewWillAppear:animated];
-    }
-    else {
-        [self.view endEditing:YES];
-        [self didSelectPost];
-    }
+- (void) viewWillAppear:(BOOL)animated {
+  [self didSelectPost];
+  [super viewWillDisappear:animated];
```

Comment:
`viewWillAppear` should call `[super viewWillAppear:..]` also calling parent objects should be kept as logical as possible to avoid any issue in the view lifecycle:
- will/didAppear/viewDidLoad should call super before doing anyhing else
- willDissapear should call super after doing everything else